### DISCUSSION TITLE
slicing arrays is not working

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [pmndrs/zustand#2219] [Q&A]
I have this state setter, I always have more than 2 elements in history, am I missing something?
```ts
setHistory: (value)=> {
     set((s) => {
         const history = [...(s.history), { value }].slice(-2) as { value: string }[]
         return { history };
     }
}
```

**Comment by dbritto-dev** [comment]
@Ali-Hussein-dev would you mind sharing us a minimal repro on codesandbox? I would like to help
